src/cmds/fs/sort.c

```diff
@@ -0,0 +1,173 @@
+/**
+ * @file
+ * @brief Sort lines of text files to standart output.
+ *
+ * @date 6.10.18
+ * @author Mark Kovalev
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <unistd.h>
+#include <errno.h>
+
+static int blank, ign_case, reverse;
+
+static void print_usage(void) {
+	printf(
+		"Usage: sort [OPTION]... [FILE]...\n"
+		"Write sorted lines of FILE to standard output.\n"
+		"\nOrdering options:\n\n"
+		"  -b\tignore leading blanks\n"
+		"  -f\tfold lower case to upper case characters\n"
+		"  -r\treverse the result of comparisons\n"
+		"\nOther options:\n\n"
+		"  -h\tdisplay this help and exit\n"
+		"\n*** WARNING ***\n"
+		"The locale specified by the environment affects sort order.\n"
+		"This implementation assumes that LC_ALL=C set to get the\n"
+		"traditional sort order that uses native byte values.\n"
+	);
+}
+
+/* 	This function similar to "strcasecmp" except for "tolower" changed to "toupper",
+	because "sort -f" folds lower case to upper case characters */
+static int case_compare(const char *str1, const char *str2) {
```

Comment:
I figured out that `strcasecmp` still can't do the job well with `-f` option, when there're characters between 'Z' and 'a' (like '[', '\\', ']' and so on). GNU `sort -f` treats `'z' < '['` and `'A' < '['`, while for `strcasecmp` `'[' < 'z'`  and `'[' < 'A'`. It happens because `strcasecmp` folds chars to lower case, while GNU sort folds chars to upper case. So I've rewrited it with just changing `tolower` to `toupper`.

Reply:
Yes, it's more correct.